Source/RxCocoa/UIScrollView+reachedBottom.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  UIScrollView+reachedBottom.swift
+//  RxSwiftExt
+//
+//  Created by Anton Nazarov on 09/05/2019.
+//  Copyright © 2019 RxSwift Community. All rights reserved.
+//
+
+import UIKit
+import RxSwift
+import RxCocoa
+
+public extension Reactive where Base: UIScrollView {
+    /**
+     Shows if the bottom of the UIScrollView is reached.
+     - parameter offset: A threshhold indicating the bottom of the UIScrollView.
+     - returns: ControlEvent that emits when the bottom of the base UIScrollView is reached.
+     */
+    func reachedBottom(offset: CGFloat = 0.0) -> ControlEvent<Void> {
+        let source = contentOffset.map { contentOffset in
+            let visibleHeight = self.base.frame.height - self.base.contentInset.top - self.base.contentInset.bottom
```

Comment:
I know we've had this discussion already below but - I still think it's best to be safe and add a `[weak base]` or `[weak self]` here.

The fact you don't have a retain cycle is related to the current implementation of `contentOffset`. There's no guarantee a retain cycle won't be create if the underlying implementation changes in a later version of RxSwift. 

My opinion is that external code should protect itself and not count on the core code. WDYT?

Reply:
> external code should protect itself and not count on the core code

I think it's against all principle of abstraction. If core changing -> you change your system, which is based on the core. This is how things work from my experience. We don't wrap swift core functions in our own, because "one day API will be changed, so we can't trust swift"

If you want, I can add a unit test on the reference cycle here. It will fail if `contentOffset` changes.

Reply:
Sorry, but I fail to understand why you're so stubborn about this specific thing. There is no performance gain, just added safety.

When you wrap Swift, you're wrapping a black box. You have no idea to know what's going on inside to know if the inner code keeps you safe from a retain cycle or not - so when wrapping 3rd party code, you should treat it as that same black box. 

It's somewhat bad-practice to base an implementation detail related to memory safety only on the fact you're intimate with the specific implementation details of the wrapped implementation.

It doesn't make too much sense to me. If there was some performance hit here or something, I'd definitely at least understand where you're coming from, but here it just seems you don't want to add it for now real apparent reason.

Reply:
So, you won't merge it, if i won't add `[weak self]`?

Reply:
This isn't your repo, it's a community repo, everyone can make modifications. 

Even if I'd merge this code, it doesn't mean I can't add the capture group later. 

Regardless, you seem to be "fighting" with myself and @jdisho on every single comment we write, after basically taking over this PR which was originally written by @jdisho. 

Contributing to OSS is awesome, and I'm a huge fan of people who do so such as yourself, but you should also be a good OSS citizen. In this specific case you're fighting over nothing.

I prefer the added safety and taking care of "our own", which is not useless, even if you think so. UIKit implementation also changed and leaked several times, we need to do what we can so the code is independent in that case.

Reply:
Fixed, thanks